src/ShopBundle/objects/db/TShopOrder.class.php

```diff
@@ -825,8 +873,16 @@ public function GetNetProductValue()
         $aTaxValueGroups = array();
         while ($oArticle = $oArticles->Next()) {
             if (!array_key_exists($oArticle->fieldVatPercent, $aTaxValueGroups)) {
+                /**
+                 * @psalm-suppress InvalidArrayOffset
+                 * @FIXME Float array keys are converted to integers - this could lead to unwanted behaviour
+                 */
                 $aTaxValueGroups[$oArticle->fieldVatPercent] = 0;
```

Comment:
This is a very well hidden bug: When using floats as an array key, they are implicitly casted to integers. In this case the array keys are used further down in this method in order to load `TdbShopVat` which means that part of the code will fail for non-integer vat percentages.